Control/public/about/About.js

```diff
@@ -0,0 +1,176 @@
+   * @returns {void}
+   */
+  async retrieveInfo() {
+    for (const key in this.services) {
+      this.retrieveServiceState(key, this.services[key])
+    }
+    this.retrieveWsInfo();
+  }
+
+  /**
+   * Return a list of services that are in error state being either:
+   * * RemoteData.failure - case in which the name and message is returned only
+   * * Service in configured state but in failure - case in which full payload is returned
+   * @returns {Array<{name, message}|Service>} - list of services
+   */
+  getInErrorServices() {
+    return Object.entries(this.statuses)
+      .filter(([_, value]) => {
+        if (value.isFailure()) {
+          return true;
+        } else if (value.isSuccess()) {
+          const {status: {ok, configured}} = value.payload;
+          return configured && !ok;
+        }
+        return false;
+      })
+      .map(([name, value]) => {
+        if (value.isFailure()) {
+          return {
+            name,
+            message: this.statuses[name].payload.message
+          };
+        } else {
+          return value.payload;
+        }
+      });
+  }
+
+  /**
+   * Return a list of names of components in RemoteData.loading state
+   * @returns {Array<string>}
+   */
+  getInLoadingComponents() {
+    return Object.entries(this.statuses)
+      .filter(([_, value]) => Boolean(value.isLoading()))
+      .map(([name, _]) => name);
+  }
+
+  /**
+   * Method to retrieve a list of Services from components in RemoteData.success and status.ok
+   * @returns {Array<Service>}
+   */
+  getInSuccessServices() {
+    return Object.entries(this.statuses)
+      .filter(([_, value]) => Boolean(value.isSuccess()))
+      .map(([_, value]) => value.payload)
+      .filter((service) => service.status.ok & service.status.configured);
+  }
+  /**
+   * Method to retrieve a list of components that are not enabled
+   * @returns {Array<Service>}
+   */
+  getNotEnabledServices() {
+    const t = Object.entries(this.statuses)
+      .filter(([_, value]) => Boolean(value.isSuccess()))
+      .filter(([_, value]) => !value.payload.status.configured)
+      .map(([name, value]) => ({name, ...value.payload}))
+    return t;
+  }
+
+  /**
+   * Method to retrieve the status of a service given its key and path to query;
+   * If the service is part of AliECS integrated services, then:
+   * * status of previous integrated services is removed
+   * * a general loading state is added as we are not aware what services we will retrieve
+   * * on success, all services are added back into the status map
+   * @param {string} key
+   * @param {string} path - e.g. core/services
+   */
+  async retrieveServiceState(key, path) {
+    if (key === INTEGRATED_SERVICE_LABEL) {
+      Object.keys(this.statuses)
+        .filter((name) => name.startsWith(INTEGRATED_SERVICE_LABEL))
+        .forEach((name) => delete this.statuses[name]); // remove status of existing components in case they were removed in the meantime
+    }
```

Comment:
I think you should do this after putting in loading state (it will not change a lot but I think it makes more sense

Reply:
Updated